src/Utility/Hash.php

```diff
@@ -1070,7 +1064,7 @@ public static function sort(array $data, string $path, $dir = 'asc', $type = 're
      * @param mixed $key The key for the data.
      * @return array
      */
-    protected static function _squash(array $data, $key = null): array
+    protected static function _squash(array $data, mixed $key = null): array
```

Comment:
Perhaps using `int|string` instead of `mixed` for `$key` might make static analyzers happy.